lib/modules/settings/models/notification_settings.dart

```diff
@@ -25,40 +33,69 @@ class NotificationRingtone {
   });
 }
 
-class PrayerNotificationSettings {
-  PrayerNotificationSettings(
-      {required this.athanSettings, required this.iqamaSettings});
+class PreAthanSettings {
+  PreAthanSettings({
+    required this.enabled,
+    required this.notificationSettings,
+    required this.before,
+  });
 
-  PrayerNotificationSettings.defaultValue()
-      : this(
-          athanSettings: NotificationSettings(
-            sound: true,
-            vibration: false,
-            ringtone: athanRingtones[0],
-          ),
-          iqamaSettings: IqamaSettings(
-            enabled: true,
-            delay: 10,
-            notificationSettings: NotificationSettings(
-              sound: true,
-              vibration: false,
-              ringtone: iqamaRingtones[0],
-            ),
-          ),
-        );
+  bool enabled;
+  NotificationSettings notificationSettings;
+  int before;
 
-  NotificationSettings athanSettings;
-  IqamaSettings iqamaSettings;
+  factory PreAthanSettings.defaultValue() => PreAthanSettings(
+        enabled: false,
+        notificationSettings: NotificationSettings(
+          sound: true,
+          vibration: false,
+          ringtone: preAthanRingtones[0],
+        ),
+        before: 15,
```

Comment:
isn't 15 too long?
I don't mind it since they can change, but for me 5 seems more reasonable, since its mostly for preparing for prayer

Reply:
Its the default value but the range is 5-120min

Reply:
lets keep this for now until we get review, what ppl want